charm/cloudinit-juju-3.1.yaml

```diff
@@ -0,0 +1,162 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Easiest way to deploy NetBox. The only requirement is multipass.
+
+# To deploy:
+# multipass launch  -vvvv --cloud-init cloudinit-juju-3.1.yaml --timeout 1800 --name netbox --memory 4G --cpus 3 --disk 30G 22.04
+# Or from the cloud init in the repo (main branch):
+# multipass launch  -vvvv --cloud-init https://raw.githubusercontent.com/canonical/netbox/main/charm/cloudinit-juju-3.1.yaml --timeout 1800 --name netbox --memory 4G --cpus 3 --disk 30G 22.04
+
+# After deploying, you can get the url for NetBox, that will work in the host computer with:
+# multipass exec netbox -- juju run traefik-k8s/0 show-proxied-endpoints --format=yaml
+# It will be something like http://<IP of the VM>/netbox-netbox
+
+# And you can create an admin user with:
+# multipass exec netbox -- juju run netbox/leader create-superuser username=admin email=netbox@example.com
+# The response will include the password. You can now login into NetBox with that user.
+
+# When you are finished, you can delete the multipass VM with:
+# multipass delete netbox --purge
+
+# To debug possible problems, you can inspect the next log files:
+# multipass exec netbox -- tail -f /var/log/cloud-init-output.log -f
+# multipass exec netbox -- tail -f /var/log/cloud-init.log -f
+
+package_update: true
+
+packages:
+- jq
+- gnome-keyring
+- sshfs
+- docker.io
+
+snap:
+  commands:
+  - snap install juju --channel=3.1/stable
+  - snap install microk8s --channel 1.29-strict/stable
+  - snap alias microk8s.kubectl kubectl
+  - snap alias microk8s.kubectl k
+  - snap install yq
+  - snap install multipass-sshfs
+
+runcmd:
+
+- |
+  # disable swap
```

Comment:
Out of curiosity: Why is this needed?

Reply:
That's is basically a remnant from the charm-dev blueprint. In any case, it is probably a good practice, as I'd rather have process crashing that using swap memory...

Reply:
Interesting. I ended up digging into the question and found this issue https://github.com/kubernetes/kubernetes/issues/53533